smarts/core/smarts.py

```diff
@@ -1385,13 +1385,17 @@ def _try_emit_envision_state(self, provider_state: ProviderState, obs, scores):
                 veh_type = (
                     v.vehicle_config_type if v.vehicle_config_type else v.vehicle_type
                 )
-                traffic[v.vehicle_id] = envision_types.TrafficActorState(
-                    actor_type=envision_types.TrafficActorType.SocialVehicle,
-                    vehicle_type=veh_type,
-                    position=tuple(v.pose.position),
-                    heading=float(v.pose.heading),
-                    speed=v.speed,
-                )
+
+                SUPPORTED_VEHICLES = {"car", "truck", "passenger", "bus", "trailer", "coach"}
+
+                if veh_type in SUPPORTED_VEHICLES:
```

Comment:
I am not entirely confident in skipping any default without some kind of logging. The best option would be to pass on the vehicle type even if it is wrong and then log it on envision's side.

Reply:
how should we prevent the unsupported types from causing issues though? Currently passenger, motorcycle and pedestrian aren't supported

Reply:
Actually (although I just approved this PR), I agree with Tucker here.  I think this is best handled on the Envision side.   What I would do there is if type is unsupported (either "motorcycle" or "pedestrian"), I would log a warning (once per vehicle id) and then try to display a simple (small) "box" at the position specified.

Reply:
After some digging it seems as though pedestrian and motorcycle are supported by envision after all:

https://github.com/huawei-noah/SMARTS/blob/develop/envision/web/src/render_helpers.js

Should I just add the missing types to `envision/types.py`?

Reply:
Yeah, I think so.